merge/multiple_appliers_test.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merge_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/fieldpath"
+	. "sigs.k8s.io/structured-merge-diff/internal/fixture"
+	"sigs.k8s.io/structured-merge-diff/merge"
+)
+
+func TestMultipleAppliersSet(t *testing.T) {
+	tests := map[string]TestCase{
+		"remove_one": {
+			Ops: []Operation{
+				Apply{
+					Manager:    "apply-one",
+					APIVersion: "v1",
+					Object: `
+						list:
+						- name: a
+						- name: b
+					`,
+				},
+				Apply{
+					Manager:    "apply-two",
+					APIVersion: "v2",
+					Object: `
+						list:
+						- name: c
+					`,
+				},
+				Apply{
+					Manager:    "apply-one",
+					APIVersion: "v3",
+					Object: `
+						list:
+						- name: a
+					`,
+				},
+			},
+			Object: `
+				list:
+				- name: a
+				- name: c
+			`,
+			Managed: fieldpath.ManagedFields{
+				"apply-one": &fieldpath.VersionedSet{
+					Set: _NS(
+						_P("list", _KBF("name", _SV("a")), "name"),
```

Comment:
And so we're considering also having `_P("list", _KBF("name", _SV("a")))` here, right?

Reply:
Yes. When and if we make that change, we will have to change all of our tests.